jung-algorithms/src/main/java/edu/uci/ics/jung/layout/model/AbstractLayoutModel.java

```diff
@@ -35,11 +35,48 @@
   protected CompletableFuture theFuture;
   protected LayoutModel.ChangeSupport changeSupport = new DefaultLayoutModelChangeSupport();
 
-  public AbstractLayoutModel(Graph<N> graph, PointModel<P> pointModel, int width, int height) {
-    this(graph, pointModel, width, height, 0);
+  public static class Builder<N, P> {
+    protected Graph<N> graph;
+    protected PointModel<P> pointModel;
+    protected int width;
+    protected int height;
+    protected int depth;
+
+    public AbstractLayoutModel.Builder<N, P> setGraph(Graph<N> graph) {
+      this.graph = graph;
+      return this;
+    }
+
+    public AbstractLayoutModel.Builder<N, P> setPointModel(PointModel<P> pointModel) {
+      this.pointModel = pointModel;
+      return this;
+    }
+
+    public AbstractLayoutModel.Builder<N, P> setSize(int width, int height, int depth) {
```

Comment:
or maybe add the separate methods, the way you can create a Dimension with (width,height) or separate setters. Funny that I recently broke my sample program with a vertical Label because I 'fixed' the intentional swapping of width and height.

Reply:
this right there is why I want to just use the separate methods :)

Reply:
It started as a refactoring to replace setSize(java.awt.Dimension d) with something non-awt